### PR TITLE
fix: complete changes for CheckSubstituteAndUpdateState

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/core/02-client/keeper/events.go [hunk 1]
```diff
@@ -82,13 +82,12 @@ func EmitUpgradeClientEvent(ctx sdk.Context, clientID string, clientState export
...
 
 // EmitUpdateClientProposalEvent emits an update client proposal event
-func EmitUpdateClientProposalEvent(ctx sdk.Context, clientID string, clientState exported.ClientState) {
+func EmitUpdateClientProposalEvent(ctx sdk.Context, clientID, clientType string) {
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
 			types.EventTypeUpdateClientProposal,
 			sdk.NewAttribute(types.AttributeKeySubjectClientID, clientID),
-			sdk.NewAttribute(types.AttributeKeyClientType, clientState.ClientType()),
-			sdk.NewAttribute(types.AttributeKeyConsensusHeight, clientState.GetLatestHeight().String()),
```
Comment (colin-axner) [inline modules/core/02-client/keeper/events.go:91]: There's no way to guarantee the latest height of the substitute is the latest height of subject. I question the usefulness of this information, so I'd prefer to remove it for now until it is requested (at which time we can query the subject client state)

File: modules/core/02-client/keeper/proposal.go [hunk 2]
```diff
@@ -44,27 +44,26 @@ func (k Keeper) ClientUpdateProposal(ctx sdk.Context, p *types.ClientUpdatePropo
...
 	}
 
-	clientState, err := subjectClientState.CheckSubstituteAndUpdateState(ctx, k.cdc, subjectClientStore, substituteClientStore, substituteClientState)
-	if err != nil {
+	if err := subjectClientState.CheckSubstituteAndUpdateState(ctx, k.cdc, subjectClientStore, substituteClientStore, substituteClientState); err != nil {
 		return err
 	}
 
-	k.Logger(ctx).Info("client updated after governance proposal passed", "client-id", p.SubjectClientId, "height", clientState.GetLatestHeight().String())
+	k.Logger(ctx).Info("client updated after governance proposal passed", "client-id", p.SubjectClientId, "height")
```
Comment (colin-axner) [inline modules/core/02-client/keeper/proposal.go:51]:
```suggestion
	k.Logger(ctx).Info("client updated after governance proposal passed", "client-id", p.SubjectClientId)
```

File: modules/core/02-client/keeper/proposal.go [hunk 3]
```diff
@@ -44,27 +44,26 @@ func (k Keeper) ClientUpdateProposal(ctx sdk.Context, p *types.ClientUpdatePropo
...
-	k.Logger(ctx).Info("client updated after governance proposal passed", "client-id", p.SubjectClientId, "height", clientState.GetLatestHeight().String())
+	k.Logger(ctx).Info("client updated after governance proposal passed", "client-id", p.SubjectClientId)
 
 	defer func() {
 		telemetry.IncrCounterWithLabels(
 			[]string{"ibc", "client", "update"},
 			1,
 			[]metrics.Label{
-				telemetry.NewLabel(types.LabelClientType, clientState.ClientType()),
+				telemetry.NewLabel(types.LabelClientType, substituteClientState.ClientType()),
```
Comment (colin-axner) [inline modules/core/02-client/keeper/proposal.go:58]: cannot guarantee client type doesn't change for the subject, so we use the substitute client type

File: modules/core/exported/client.go [hunk 4]
```diff
@@ -69,8 +69,9 @@ type ClientState interface {
 	// Upon successful update, a list of consensus heights is returned. It assumes the ClientMessage has already been verified.
 	UpdateState(sdk.Context, codec.BinaryCodec, sdk.KVStore, ClientMessage) []Height
 
-	// Update and Misbehaviour functions
-	CheckSubstituteAndUpdateState(ctx sdk.Context, cdc codec.BinaryCodec, subjectClientStore, substituteClientStore sdk.KVStore, substituteClient ClientState) (ClientState, error)
+	// CheckSubstituteAndUpdateState must verify that the provided substitute may be used to update the subject client.
+	// The light client must set the updated client and consensus states within the clientStore for the subject client.
+	CheckSubstituteAndUpdateState(ctx sdk.Context, cdc codec.BinaryCodec, subjectClientStore, substituteClientStore sdk.KVStore, substituteClient ClientState) error
```
Comment (colin-axner) [inline modules/core/exported/client.go:74]: I realized we (I) forgot to update the interface for this function

**Review by damiannolan — APPROVED** [review]
Good spot! LGTM

**Review by seantking — APPROVED** [review]

**Pull request merged**